tools/test/toolchains/test_toolchains.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python3
+# Copyright (c) 2019 Arm Limited and Contributors. All rights reserved.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+"""Test the arm toolchain."""
+
+import os
+import sys
+
+import mock
+
+
+ROOT = os.path.abspath(
+    os.path.join(os.path.dirname(__file__), "..", "..", "..")
+)
+sys.path.insert(0, ROOT)
+
+from tools.toolchains.arm import ARM_STD, ARM_MICRO, ARMC6
+from tools.toolchains.gcc import GCC_ARM
+from tools.toolchains.iar import IAR
+
+
+class TestArmToolchain:
+    """Test Arm classes."""
+
+    def test_arm_minimal_printf(self):
+        """Test that linker flags are correctly added to an instance of ARM."""
+        mock_target = mock.MagicMock()
+        mock_target.core = "Cortex-M4"
+        mock_target.printf_lib = "minimal-printf"
+        mock_target.supported_toolchains = ["ARM", "uARM", "ARMC5"]
+
+        arm_std_obj = ARM_STD(mock_target)
+        arm_micro_obj = ARM_MICRO(mock_target)
+        arm_c6_obj = ARMC6(mock_target)
+
+        assert "-DMBED_MINIMAL_PRINTF" in arm_std_obj.flags["common"]
```

Comment:
Please use https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertEqual

Reply:
You may want to also update `tools/test/toolchains/api_test.py` and `tools/test/toolchains/arm_support_test.py`

Reply:
[Done}(https://github.com/ARMmbed/mbed-os/compare/c72e2ae73a07b2e0f4c388c90f97033cab4693cd..b758988ec8e1ff6101508609587b9a83748cda1a).